zephyr/test/vmh.c

```diff
@@ -58,26 +58,37 @@ static void test_vmh_alloc_free_no_check(struct vmh_heap *heap,
 	}
 }
 
-static void verify_memory_content(void *ptr, uint32_t alloc_size)
+/* Fill memory with test pattern */
+static void verify_memory_content(void *ptr, uint32_t alloc_size, bool fill)
 {
+	const uint32_t pattern1 = (uint32_t)ptr ^ 0xDEADBEEF;
+	const uint32_t pattern2 = (uint32_t)ptr ^ 0xCAFEBABE;
+	const uint32_t pattern3 = (uint32_t)ptr ^ 0xFEEDFACE;
+
+	zassert_true(alloc_size >= 16, "alloc size is below the minimum value.");
```

Comment:
Why is there a minimal value ?

Reply:
@dabekjakub The minimum size results from using 3 patterns of 4 bytes each and taking into account rounding up position of the middle pattern. I don't see how testing 4..8 byte buffers would be better than 16..64 byte buffers.

Reply:
Btw, would minimal size be a Kconfig option or based on cache line size etc ?